src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -368,18 +369,22 @@ var ReactCompositeComponentMixin = {
    * @final
    * @internal
    */
-  unmountComponent: function() {
+  unmountComponent: function(safely) {
     if (!this._renderedComponent) {
       return;
     }
     var inst = this._instance;
 
     if (inst.componentWillUnmount) {
-      inst.componentWillUnmount();
+      if (safely) {
+        ReactErrorUtils.invokeGuardedCallback('componentWillUnmount', inst.componentWillUnmount.bind(inst));
```

Comment:
Right, but if we are logging these errors then naming the guard with a string would be helpful. I don't think concatenating is a big deal -- string operations are done lazily anyway.